uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/admin-dashboard/modules/settings.js

```diff
@@ -5,6 +5,7 @@
     'common/angular'
   ], function (angular) {
     return angular.module('admin-dashboard.settings', []).constant('settings', {
+      contactId: CRM.vars.leaveAndAbsences.loggedInUserId,
```

Comment:
While I agree with you, I am not sure if it is a good idea now, because if I change it for one component, I need to change the other components too to keep consistency so in the end it will be too huge of a task at this point of time. Also the confusion only arises for `leave-request-popup` as it has more than one contact ID to deal with, but this component should be fine. cc @AkA84